library/ecp.c

```diff
@@ -767,6 +767,63 @@ int mbedtls_ecp_point_write_binary( const mbedtls_ecp_group *grp,
     return( ret );
 }
 
+#if defined(MBEDTLS_ECP_SHORT_WEIERSTRASS_ENABLED)
+static int mbedtls_ecp_sw_derive_y( const mbedtls_ecp_group *grp,
+                                    const mbedtls_mpi *X,
+                                    mbedtls_mpi *Y,
+                                    int parity_bit )
+{
+    /* y^2 = x^3 + ax + b
+     * sqrt(w) = w^((p+1)/4) mod p   (for prime p where p = 3 mod 4) */
+
+    if( mbedtls_mpi_get_bit( &grp->P, 0 ) != 1 || /*(p = 3 mod 4)*/
```

Comment:
Style: I don't see the point of parentheses around a whole comment. Also, the comment would be better placed before the block than on each line, as it's the combination of both lines that tests whether p == 3 mod 4.

Reply:
:rocket: